tests/unit/test_contrib/test_sqlalchemy/test_repository/test_repository.py

```diff
@@ -28,30 +25,101 @@
     SearchFilter,
 )
 from litestar.contrib.sqlalchemy import base
+from litestar.contrib.sqlalchemy.repository import SQLAlchemyAsyncRepository
 from tests.helpers import maybe_async
-from tests.unit.test_contrib.test_sqlalchemy.models_uuid import (
-    AuthorAsyncRepository,
-    AuthorSyncRepository,
-    BookAsyncRepository,
-    BookSyncRepository,
-    ItemAsyncRepository,
-    ItemSyncRepository,
-    ModelWithFetchedValueAsyncRepository,
-    ModelWithFetchedValueSyncRepository,
-    RuleAsyncRepository,
-    RuleSyncRepository,
-    TagAsyncRepository,
-    TagSyncRepository,
-    UUIDAuthor,
-    UUIDBook,
-    UUIDItem,
-    UUIDModelWithFetchedValue,
-    UUIDRule,
-    UUIDTag,
-)
+from tests.unit.test_contrib.test_sqlalchemy import models_bigint, models_uuid
 
 from .helpers import update_raw_records
 
+RepositoryPKType = Literal["uuid", "bigint"]
+AuthorModel = Type[Union[models_uuid.UUIDAuthor, models_bigint.BigIntAuthor]]
+RuleModel = Type[Union[models_uuid.UUIDRule, models_bigint.BigIntRule]]
+ModelWithFetchedValue = Type[Union[models_uuid.UUIDModelWithFetchedValue, models_bigint.BigIntModelWithFetchedValue]]
+ItemModel = Type[Union[models_uuid.UUIDItem, models_bigint.BigIntItem]]
+TagModel = Type[Union[models_uuid.UUIDTag, models_bigint.BigIntTag]]
+
+AnyAuthor = Union[models_uuid.UUIDAuthor, models_bigint.BigIntAuthor]
+AuthorRepository = SQLAlchemyAsyncRepository[AnyAuthor]
```

Comment:
This *should* be `SQLAlchemyAsyncRepository[AnyAuthor] | SQLAlchemySyncRepository[AnyAuthor]`, but I can't get mypy to correctly narrow the return values of `maybe_async` then. Not sure why since it does work correctly in other places.

Anyway, this isn't that big of a deal since we inherit the typing for the async/sync mostly from SQLA and type check the repository code.